engine/buildkit/executor_spec.go

```diff
@@ -896,8 +896,14 @@ func (w *Worker) setupNestedClient(ctx context.Context, state *execState) (rerr
 	w.execMD.ClientStableID = randid.NewID()
 
 	// include SSH_AUTH_SOCK if it's set in the exec's env vars
-	if v, ok := state.origEnvMap["SSH_AUTH_SOCK"]; ok {
-		w.execMD.SSHAuthSocketPath = v
+	if sockPath, ok := state.origEnvMap["SSH_AUTH_SOCK"]; ok {
+		if homeDir, ok := state.origEnvMap["HOME"]; ok {
```

Comment:
This will result in `SSH_AUTH_SOCK` only working if `HOME` is set, even if the path doesn't have a `~` in it, right? It seems like `HOME` should only matter if there's a `~` in the `SSH_AUTH_SOCK` path.

Reply:
Corrected, thanks 🙏